src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php

```diff
@@ -345,12 +345,22 @@ public function getFieldDescriptors()
      */
     public function getById($id, $locale)
     {
+        $mediaEntity = $this->getEntityById($id);
+
+        return $this->addFormatsAndUrl(new Media($mediaEntity, $locale, null));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getEntityById($id)
```

Comment:
@danrot I need it to set the avatar for a contact on e.g a post- or a put request:
´´´
/**
     * Sets a media with a given id as the avatar of a given contact.
     *
     * @param Contact $contact
     * @param array $avatar with id property
     */
    private function setAvatar(Contact $contact, $avatar)
    {
        $mediaEntity = null;
        if (is_array($avatar) && $this->getProperty($avatar, 'id')) {
            $mediaEntity = $this->mediaManager->getEntityById($this->getProperty($avatar, 'id'));
        }
        $contact->setAvatar($mediaEntity);
    }
´´´